.github/workflows/test.yml

```diff
@@ -60,13 +60,17 @@ jobs:
       with:
         path: ~/.elm
         key: elm-${{ runner.os }}-${{ hashFiles('**/elm.json', 'elm-tooling.json') }}
+        restore-keys: |
+          elm-${{ runner.os }}-
```

Comment:
I've read https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows but I still don't get how this is not dangerous.

If `elm.json` has changed, then recovering from other caches seems like a recipe for problems, though I don't understand what it recovers. What am I missing?

Reply:
Say I've got an `elm.json` with `elm-review@2.14.1` and `elm-syntax@8.3.4`. I bump `elm-syntax`, so the old cache is invalidated, but it'll still grab the last cache from the same branch, then `main` (so you can't get a cache from a fork and get a supply-chain attack), so `elm make` doesn't have to redownload the unchanged `elm-review`. Then, it'll upload the new cache (with the new `elm-syntax`) with a new hash.
It has the potential to create larger caches, but if it becomes a problem, you can just clear the cache.